atomix/cluster/src/main/java/io/atomix/raft/roles/LeaderRole.java

```diff
@@ -1441,6 +1448,28 @@ private void safeAppendEntry(
             });
   }
 
+  /**
+   * Returns true if the supplied position is higher than the last ZeebeEntry in the log or if no
+   * ZeebeEntry was found at all.
+   */
+  private boolean isEntryConsistent(long newEntryPosition) {
+    Indexed<RaftLogEntry> lastEntry = raft.getLogWriter().getLastEntry();
+
+    if (lastEntry == null || lastEntry.type() != ZeebeEntry.class) {
+      long index = raft.getLogWriter().getLastIndex();
```

Comment:
Can you explain? If the last entry is null then `getLastIndex()` will already return the segment's starting index minus 1:
https://github.com/zeebe-io/zeebe/blob/66a4acab1cbee3eeeb2fd56675add9535cfd846e/atomix/storage/src/main/java/io/atomix/storage/journal/MappedJournalSegmentWriter.java#L85
So it seemed safer to just start there since at worst we check an index twice which is safer than skipping what might be the actual last zeebe entry. But I'm not an expert on the entry/log structure so this might not make sense though

Reply:
When can `lastEntry` be null?
As far as I know reader reset operation is not very efficient. It re-reads all entries in a segment from the beginning. So it would be good if we can avoid that extra reset that reads the same last entry.